### PR TITLE
Fix message not restored in delete operations

### PR DESCRIPTION
**Pull request by carlesarnal**

### CODE REVIEW

**Comment by carlesarnal** [comment]
@EricWittmann this fixes the linked issue. Let me know if it's the proper way of solving it or if I need to do something different

File: src/test/resources/fixtures/cmd/commands/delete-operation/asyncapi-2/delete-operation.before.json [hunk 1]
```diff
@@ -8,7 +8,10 @@
     "myChannel": {
       "description": "This is myChannel description",
       "subscribe": {
-        "summary": "Subscribe summary"
+        "summary": "Subscribe summary",
+        "message": {
+          "name": "bullshit"
```
Comment (EricWittmann) [inline src/test/resources/fixtures/cmd/commands/delete-operation/asyncapi-2/delete-operation.before.json:13]: Can we use less offensive test data?  :)

File: src/main/java/io/apicurio/datamodels/asyncapi/io/AaiDataModelWriter.java [hunk 2]
```diff
@@ -390,8 +390,13 @@ public void visitOperation(Operation node) {
...
 
-        JsonCompat.setPropertyNull(json, Constants.PROP_TRAITS); // prop
-        JsonCompat.setPropertyNull(json, Constants.PROP_MESSAGE); // prop
+        JsonCompat.setPropertyNull(json, Constants.PROP_TRAITS);
+        if (aaiNode.message != null) {
+            Object messageJson = writeMessageBase(aaiNode.message);
+            JsonCompat.setProperty(json, Constants.PROP_MESSAGE, messageJson);
+        } else {
+            JsonCompat.setPropertyNull(json, Constants.PROP_MESSAGE);
+        }
```
Comment (EricWittmann) [inline src/main/java/io/apicurio/datamodels/asyncapi/io/AaiDataModelWriter.java:399]: So I think the way this should work is that the `message` child of the operation should be processed in a separate visit method called `visitMessage`.  The reason we set nulls for the child property values at this level is to force the ordering of the properties in the resulting JSON.  

So if there's a bug here, it's likely to be in the `visitMessage` method.

File: src/test/resources/fixtures/cmd/commands/delete-operation/asyncapi-2/delete-operation.before.json (hunk 1 above)
Comment (carlesarnal) [reply src/test/resources/fixtures/cmd/commands/delete-operation/asyncapi-2/delete-operation.before.json:13]: Sorry, copied from the issue test data.

File: src/main/java/io/apicurio/datamodels/asyncapi/io/AaiDataModelWriter.java (hunk 2 above)
Comment (carlesarnal) [reply src/main/java/io/apicurio/datamodels/asyncapi/io/AaiDataModelWriter.java:399]: Ok, I will revisit this and let you know my findings.

**Comment by EricWittmann** [comment]
@carlesarnal reminder that this is still pending. :)

**Comment by carlesarnal** [comment]
> @carlesarnal reminder that this is still pending. :)

Yes, sorry, it's on my to-do list but other priorities arose. Will revisit this asap.

**Comment by EricWittmann** [comment]
No no - not a priority.  Just making sure it was still on your backlog.  :)

**Pull request closed without merging**